## WELCOME

#### This is an introductory guide to the Riot API by Jack J from iTero Gaming

#### You can follow me on Twitter: https://twitter.com/JackJGaming
#### or check out the site: https://itero.gg

##### An introduction to the Riot API

Have you ever wondered how sites like u.gg and op.gg get their data? \
Are you looking to become an esports analyst but not sure what skills you need? \
Want to create an app or site but not sure how to build one?

All these questions can be answered the same: The Riot API

In this short introductory guide I will take you through the process of building your first ever Riot API product \
It will require some very basic Python knowledge, but I'll explain every step along the way

I hope you enjoy, 

Jack J \
Founder & Lead Data Scientist @ iTero Gaming

### Getting a key

Head over to https://developer.riotgames.com/ \
Login with your Riot Account and accept the terms of service \
At the bottom, click "I'm not a Robot" and regenerate your API key \
Copy the API key below

In [68]:
api_key = ""

### Our First API Call

Go back to the developer portal and at the top, click "APIS" \
On the left there will be a list of all the available APIs that Riot offers \
Don't be overwhelmed, it covers most of their games and we will only need a small number of them 

Scroll down until you find "Summoner-V4", once you click it you will see a list of API calls in the middle \
From here, click on the API called: /lol/summoner/v4/summoners/by-name/{summonerName} \
Scroll to the bottom of the page and type in your Summoner Name, select your Region and then select Development API Key \
Then, click "EXECUTE REQUEST" 

You should see a response code of 200, then below that the Response Body with information about your account 

Scroll back up, just below "EXECUTE REQUEST" you will see a URL that looks something like the one below \
Replace the below api_url with the one you have on the API portal


In [2]:
api_url = "https://euw1.api.riotgames.com/lol/summoner/v4/summoners/by-name/Jack%20J"

We will be managing all our API requests through the "requests" library, which should be pre-installed

In [3]:
import requests

We can then send off any API request we want, so let's send off our URL

In [4]:
requests.get(api_url)

<Response [401]>

401? 

If you return to the portal and scroll up, you'll see a 401 error means: Unauthorized \
That's because we haven't attached the API key to our request!

When dealing with APIs, we tag on arguments to the request by adding "?" to the end of the url, followed by the variable

In [5]:
api_url = api_url + '?api_key=' + api_key

In [6]:
api_url

'https://euw1.api.riotgames.com/lol/summoner/v4/summoners/by-name/Jack%20J?api_key=RGAPI-b77ee1e3-e525-4a6d-8d5c-a9938d45f9d0'

In [7]:
requests.get(api_url)

<Response [200]>

Great! Now we're getting a 200 (which means successful) \
But where's the data? \
To get it, we need to assign the request to a variable and call the JSON

In [8]:
resp = requests.get(api_url)
player_info = resp.json()
player_info

{'id': 'gkjv-w3_gkWbFFmMZkrDj2JR-rrmv4Mj172gqRO6ycmY9FY',
 'accountId': 'aD6DoTQb_RWYLAghp_mWwuwAWHOW6qRz6zgQsf2y5YIitQ',
 'puuid': 'dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA',
 'name': 'Jack J',
 'profileIconId': 918,
 'revisionDate': 1658431032354,
 'summonerLevel': 345}

And there we go! \
We've successfully sent off and retrieved our first API request

#### Match Data

That was easy, but it's not much to show-off \
The next thing we're going to do is get some historic match data 

To do this, go back to the portal and click "MATCH-V5" on the far left \
Go the option called: /lol/match/v5/matches/by-puuid/{puuid}/ids \
As we can see from the "{puuid}" above, it needs the accounts puuid - luckily we go that from the previous API!

In [9]:
puuid = player_info['puuid']
puuid

'dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA'

Scroll down on the portal until you see the box asking for the puuid \
Copy and paste the puuid above into the "puuid" box \
Leave the Query Parameters for now but update the REGION & then APP TO EXECUTE AGAINST to Development API Key \
Execute Request!

You should see a response code of 200, and the body contains a list of keys \
Let's copy the REQUEST URL from the portal so we can get everything into Python

In [10]:
api_url = "https://europe.api.riotgames.com/lol/match/v5/matches/by-puuid/dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA/ids?start=0&count=20"

If you inspect the URL, you'll see there's already arguments at the end: "?start=0&count=20"\
In order to add another argument (your API key), we'll use the "&" symbol

In [11]:
#                    & instead of ?, because there's already a ? in the original
api_url = api_url + '&api_key=' + api_key 
api_url

'https://europe.api.riotgames.com/lol/match/v5/matches/by-puuid/dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA/ids?start=0&count=20&api_key=RGAPI-b77ee1e3-e525-4a6d-8d5c-a9938d45f9d0'

Now, the same as before - let's get the data from this url

In [12]:
resp = requests.get(api_url)
match_ids = resp.json()

match_ids

['EUW1_5982943551',
 'EUW1_5982849305',
 'EUW1_5982852241',
 'EUW1_5982834160',
 'EUW1_5982385567',
 'EUW1_5982421620',
 'EUW1_5981636284',
 'EUW1_5981413357',
 'EUW1_5981269044',
 'EUW1_5981244521',
 'EUW1_5980174276',
 'EUW1_5980007970',
 'EUW1_5979817728',
 'EUW1_5979851539',
 'EUW1_5979685858',
 'EUW1_5979617134',
 'EUW1_5978185445',
 'EUW1_5978027389',
 'EUW1_5977260536',
 'EUW1_5977193266']

This API contains a list of the match IDs from our accounts last 20 games 

Let's take the first one of these

In [13]:
recent_match = match_ids[0]
recent_match

'EUW1_5982943551'

If we return to the portal, scroll back up and open this API: "/lol/match/v5/matches/{matchId}"

Scroll down and copy the match ID into the "matchId" input, again selecting the right region & app

If everything goes well, you'll see our trusty 200 response code and a lot of data in RESPONSE BODY

Once again, copy and paste the REQUEST URL

In [14]:
api_url = "https://europe.api.riotgames.com/lol/match/v5/matches/EUW1_5982385567"

In [15]:
# This one doesn't have any arguments (see, there's no "?" above) so we'll add it with the api key
api_url = api_url + '?api_key=' + api_key

In [16]:
resp = requests.get(api_url)
match_data = resp.json()

match_data

{'metadata': {'dataVersion': '2',
  'matchId': 'EUW1_5982385567',
  'participants': ['EBIKCKjFcf6JupjkbOFUMhROmuyHeL3WWV_s1IMwNGZPDWLjdEyl_cL-l_Lsww9tbYwROnrE5c7KpQ',
   'Qs48y7lqlx8WmRxBSJUWSugwpE0YDVfMvwMkNKoH-l1_VuQhDPj1_R-1zJEEsJ31BvO9SD7hOcszzQ',
   'nySK7lWVZpFK88v5d0QYgFVvNMEb_b7RPJZfRl6-EEZQ0fGignoL9Z7AYW5G4eoXcz8xojxF0spXYQ',
   'Ew4sJmPhZ8X-wjeRmQ3bjcE5oj16vW57ocQMTx2_LzJVFRY6Vrs0j1oZbzPQsolQ9rrCWwFgce8FBA',
   'Q8HAi59kS1ejnH9d2Yddj2EDmGZjpllUx0-flyzCy4hJ-ZbakhpM9MkP4m1RUtpT-kZiuGTcxO4oJw',
   'RwaKEwuXms9HO3XWg8De88x-JXUy728vYcKjCf7B1PNxTI2VKBOUsBGO3XGbk6Q4dnazqRTSPSve4Q',
   'o3aGo6ezbFIRLMwwVFrDVNkK7qmODJfZvPGwXiBnOxOhGyHp6oYYV761L3tdUrqoHYVqIX4UcgBS7Q',
   'dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA',
   'uRGlCtWRUGdXqC6eHVT9NxjJlWLi8JUXdTG78xBIioDAUrS8j-pzGNQfrJG7pva2XELo3kujp25LBQ',
   '3G2CnK8q3T3fxzpAIztdt9qkbr9E5TREVoNpxqdPEsiL1jERXq32jhGD7tnQPz48KMkYJY0-cG5PQw']},
 'info': {'gameCreation': 1658576828000,
  'gameDuration': 2023,
 

That's a lot of data!
Let's break it down

In [17]:
# the two headline sections of the data
match_data.keys()

dict_keys(['metadata', 'info'])

In [18]:
# "metadata" contains the puuid for every player in the game
match_data['metadata']

# NOTE: The order in which these puuid's appear in this the "participants" list
# is the same order they will appear for data elsewhere in this dictionary (useful for later)

{'dataVersion': '2',
 'matchId': 'EUW1_5982385567',
 'participants': ['EBIKCKjFcf6JupjkbOFUMhROmuyHeL3WWV_s1IMwNGZPDWLjdEyl_cL-l_Lsww9tbYwROnrE5c7KpQ',
  'Qs48y7lqlx8WmRxBSJUWSugwpE0YDVfMvwMkNKoH-l1_VuQhDPj1_R-1zJEEsJ31BvO9SD7hOcszzQ',
  'nySK7lWVZpFK88v5d0QYgFVvNMEb_b7RPJZfRl6-EEZQ0fGignoL9Z7AYW5G4eoXcz8xojxF0spXYQ',
  'Ew4sJmPhZ8X-wjeRmQ3bjcE5oj16vW57ocQMTx2_LzJVFRY6Vrs0j1oZbzPQsolQ9rrCWwFgce8FBA',
  'Q8HAi59kS1ejnH9d2Yddj2EDmGZjpllUx0-flyzCy4hJ-ZbakhpM9MkP4m1RUtpT-kZiuGTcxO4oJw',
  'RwaKEwuXms9HO3XWg8De88x-JXUy728vYcKjCf7B1PNxTI2VKBOUsBGO3XGbk6Q4dnazqRTSPSve4Q',
  'o3aGo6ezbFIRLMwwVFrDVNkK7qmODJfZvPGwXiBnOxOhGyHp6oYYV761L3tdUrqoHYVqIX4UcgBS7Q',
  'dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA',
  'uRGlCtWRUGdXqC6eHVT9NxjJlWLi8JUXdTG78xBIioDAUrS8j-pzGNQfrJG7pva2XELo3kujp25LBQ',
  '3G2CnK8q3T3fxzpAIztdt9qkbr9E5TREVoNpxqdPEsiL1jERXq32jhGD7tnQPz48KMkYJY0-cG5PQw']}

In [19]:
# "info" contains lots of data about the game, like when it was created, how long it lasted etc...
match_data['info'].keys()

dict_keys(['gameCreation', 'gameDuration', 'gameEndTimestamp', 'gameId', 'gameMode', 'gameName', 'gameStartTimestamp', 'gameType', 'gameVersion', 'mapId', 'participants', 'platformId', 'queueId', 'teams', 'tournamentCode'])

In [20]:
# For instance, this is how long the game lasted:
match_data['info']['gameDuration'] / 60 # / 60 to get it into minutes

33.71666666666667

In [21]:
# Within info, participants contains a list of length 10, each with further information about the player
len(match_data['info']['participants'])

10

In [22]:
# To save time, we'll assign a variable for first player
player_data = match_data['info']['participants'][0]
player_data

{'assists': 12,
 'baronKills': 0,
 'basicPings': 8,
 'bountyLevel': 1,
 'challenges': {'12AssistStreakCount': 0,
  'abilityUses': 614,
  'acesBefore15Minutes': 0,
  'alliedJungleMonsterKills': 19.000000059604645,
  'baronBuffGoldAdvantageOverThreshold': 1,
  'baronTakedowns': 0,
  'blastConeOppositeOpponentCount': 0,
  'bountyGold': 300,
  'buffsStolen': 0,
  'completeSupportQuestInTime': 0,
  'controlWardTimeCoverageInRiverOrEnemyHalf': 0.031814515254701504,
  'controlWardsPlaced': 1,
  'damagePerMinute': 465.52874689616283,
  'damageTakenOnTeamPercentage': 0.24816975269910102,
  'dancedWithRiftHerald': 0,
  'deathsByEnemyChamps': 10,
  'dodgeSkillShotsSmallWindow': 3,
  'doubleAces': 0,
  'dragonTakedowns': 2,
  'earliestBaron': 1858.1973595,
  'earliestDragonTakedown': 1544.0788287999999,
  'earliestElderDragon': 1975.7181031,
  'earlyLaningPhaseGoldExpAdvantage': 0,
  'effectiveHealAndShielding': 0,
  'elderDragonKillsWithOpposingSoul': 0,
  'elderDragonMultikills': 0,
  'enemyCham

In [23]:
# Then, we can find out information about them in the game, like which Champion they're playing...
player_data['championName']

'Riven'

In [24]:
# Maybe their KDA?
k = player_data['kills']
d = player_data['deaths']
a = player_data['assists']
print("Kills:", k)
print("Deaths:", d)
print("Assists:", a)
print("KDA:", (k + a) / d)

Kills: 4
Deaths: 10
Assists: 12
KDA: 1.6


In [25]:
# or their role?
player_data['teamPosition']

'TOP'

However we don't want the information about any random player in that game! \
We want to get information about the player we were searching for originally, so we need to find out where they are 

To do this, we grab the list of all the participants puuids and then we use "index" to find out where our player is

In [26]:
# A list of all the participants puuids
participants = match_data['metadata']['participants']
# Now, find where in the data our players puuid is found
player_index = participants.index(puuid)
player_index

7

In [27]:
# This should match the puuid we used to search for the match IDs, go back up and check
participants[player_index]

'dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA'

Let's check it works by using this index to get the players name from the data

In [28]:
# Hopefully the name below is what you inputted into the first function!
match_data['info']['participants'][player_index]['summonerName']

'Jack J'

Now we know where our player is, we can grab their data

In [29]:
player_data = match_data['info']['participants'][player_index]

champ = player_data['championName']
k = player_data['kills']
d = player_data['deaths']
a = player_data['assists']
win = player_data['win']

print("Champ:", champ, "Kills:", k, "Deaths:", d, "Assists:", a, "Win:", win)

Champ: Ekko Kills: 12 Deaths: 7 Assists: 9 Win: False


### Gathering Data

For whatever we're building, we may want to gather lots of data about a player \
To make this easier, we'll build some functions
Let's start by looking at the API calls we've already made

In [30]:
# The first function simply gets the puuid, given a summoner name and region
# This is exactly the same as our first example, except we're building the API URL from scratch
def get_puuid(summoner_name, region, api_key):
    api_url = (
        "https://" + 
        region +
        ".api.riotgames.com/lol/summoner/v4/summoners/by-name/" +
        summoner_name +
        "?api_key=" +
        api_key
    )
    
    print(api_url)
    
    resp = requests.get(api_url)
    player_info = resp.json()
    puuid = player_info['puuid']
    return puuid  

In [31]:
summoner_name = 'Jack J'
region = 'euw1'

puuid = get_puuid(summoner_name, region, api_key)
puuid

https://euw1.api.riotgames.com/lol/summoner/v4/summoners/by-name/Jack J?api_key=RGAPI-b77ee1e3-e525-4a6d-8d5c-a9938d45f9d0


'dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA'

In [32]:
# The function to get a list of all the match IDs (2nd example above) given a players puuid and mass region
def get_match_ids(puuid, mass_region, api_key):
    api_url = (
        "https://" +
        mass_region +
        ".api.riotgames.com/lol/match/v5/matches/by-puuid/" +
        puuid + 
        "/ids?start=0&count=20" + 
        "&api_key=" + 
        api_key
    )
    
    print(api_url)
    
    resp = requests.get(api_url)
    match_ids = resp.json()
    return match_ids      

In [33]:
# NOTE: region and mass_region are different
# for instance, NA1 is the North American region
# which is part of the AMERICAS mass region
# EUW1 is Europe West region, part of the EUROPE mass region
mass_region = 'EUROPE'

match_ids = get_match_ids(puuid, mass_region, api_key)
match_ids

https://EUROPE.api.riotgames.com/lol/match/v5/matches/by-puuid/dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA/ids?start=0&count=20&api_key=RGAPI-b77ee1e3-e525-4a6d-8d5c-a9938d45f9d0


['EUW1_5982943551',
 'EUW1_5982849305',
 'EUW1_5982852241',
 'EUW1_5982834160',
 'EUW1_5982385567',
 'EUW1_5982421620',
 'EUW1_5981636284',
 'EUW1_5981413357',
 'EUW1_5981269044',
 'EUW1_5981244521',
 'EUW1_5980174276',
 'EUW1_5980007970',
 'EUW1_5979817728',
 'EUW1_5979851539',
 'EUW1_5979685858',
 'EUW1_5979617134',
 'EUW1_5978185445',
 'EUW1_5978027389',
 'EUW1_5977260536',
 'EUW1_5977193266']

In [34]:
# From a given match ID and mass region, get the data about the game
def get_match_data(match_id, mass_region, api_key):
    api_url = (
        "https://" + 
        mass_region + 
        ".api.riotgames.com/lol/match/v5/matches/" +
        match_id + 
        "?api_key=" + 
        api_key
    )
    
    resp = requests.get(api_url)
    match_data = resp.json()
    return match_data     

In [35]:
match_id = match_ids[0]
match_data = get_match_data(match_id, mass_region, api_key)
match_data

{'metadata': {'dataVersion': '2',
  'matchId': 'EUW1_5982943551',
  'participants': ['ncyjo4fLC0BOaUfrfFIglneIy70XwC3ZgZZlk7aRT9ltB879Hwwx5StYfL5VGMqZMM7gQigmaDRDTQ',
   'l3ET8BefR_ZrMtHyDAz1cpcaHEuq7zgKcTNPcH6UnQTn1CcPk0uKDREAwbOM85V92JkriKx_lQG09A',
   'N77SUkELZAGcxxlXQ5ULEjhiutmwivjHoQJ7psrHCviqKp7SQrUEPQ77kkbfOAZZgf_rxtepAJrY5g',
   'jLaC5LtV7NZPYFmqY1FniLjITe-xaPHMxzvjT3SsUjRLbe0xfQETorzdaQ_tWiRCqBqm9GinSRn-Dw',
   'U1FpfjuK2R8jLeqibLRX40UwtIOOBxOAxD5oTr8evq1WJZryb-zpPWSjcRMkKODZb1-l1j2o4rH8WA',
   '1gPc-t-pbHCJZtobohqJIhSH5jR46YNgySUH9oBO9K4hjMvtK3n5iJ1tdAD0Jb8oDDSOFuCVpHT8QQ',
   'RwaKEwuXms9HO3XWg8De88x-JXUy728vYcKjCf7B1PNxTI2VKBOUsBGO3XGbk6Q4dnazqRTSPSve4Q',
   'pSJZAxeQZ-MpBOT70uWxJupou5WjQYaq6z-3jOSj0bX5fv1YBoPForLoxch5FRntMmlnFDms3_hudg',
   'dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA',
   'ojwAyf1nZWU9vXANTYFlfw8XjzQ3-ULdcb1D9JkXzwuLYdS-8-AJ2zlQwmmmmNiyKqdL542ppSYpEQ']},
 'info': {'gameCreation': 1658604183000,
  'gameDuration': 1900,
 

In [36]:
# Given the match data and a players puuid, return the data about just them
def find_player_data(match_data, puuid):
    participants = match_data['metadata']['participants']
    player_index = participants.index(puuid)
    player_data = match_data['info']['participants'][player_index]
    return player_data

In [37]:
find_player_data(match_data, puuid)

{'assists': 15,
 'baronKills': 0,
 'basicPings': 25,
 'bountyLevel': 0,
 'challenges': {'12AssistStreakCount': 0,
  'abilityUses': 269,
  'acesBefore15Minutes': 0,
  'alliedJungleMonsterKills': 12.000000059604645,
  'baronTakedowns': 0,
  'blastConeOppositeOpponentCount': 0,
  'bountyGold': 2000,
  'buffsStolen': 0,
  'completeSupportQuestInTime': 0,
  'controlWardTimeCoverageInRiverOrEnemyHalf': 0.07430512760554808,
  'controlWardsPlaced': 2,
  'damagePerMinute': 996.76319766883,
  'damageTakenOnTeamPercentage': 0.15791430794646952,
  'dancedWithRiftHerald': 0,
  'deathsByEnemyChamps': 8,
  'dodgeSkillShotsSmallWindow': 2,
  'doubleAces': 0,
  'dragonTakedowns': 1,
  'earliestDragonTakedown': 1422.3722697161857,
  'earlyLaningPhaseGoldExpAdvantage': 0,
  'effectiveHealAndShielding': 333.19091796875,
  'elderDragonKillsWithOpposingSoul': 0,
  'elderDragonMultikills': 0,
  'enemyChampionImmobilizations': 25,
  'enemyJungleMonsterKills': 1.600000023841858,
  'epicMonsterKillsNearEnemyJun

We haven't done anything new above, just put everything into functions so it's easier to call them later!

Now, let's say we want to look at our last 20 games and extract the data \
To do this, we simply loop through the list of match ids and perform the functions \
For my example, each time we'll save information about the Champion, KDA and Result

In [38]:
# We initialise an empty dictionary to store data for each game
data = {
    'champion': [],
    'kills': [],
    'deaths': [],
    'assists': [],
    'win': []
}

for match_id in match_ids:
    print(match_id)
    
    # run the two functions to get the player data from the match ID
    match_data = get_match_data(match_id, mass_region, api_key)
    player_data = find_player_data(match_data, puuid)
    
    # assign the variables we're interested in
    champion = player_data['championName']
    k = player_data['kills']
    d = player_data['deaths']
    a = player_data['assists']
    win = player_data['win']
     
    # add them to our dataset
    data['champion'].append(champion)
    data['kills'].append(k)
    data['deaths'].append(d)
    data['assists'].append(a)
    data['win'].append(win)    

EUW1_5982943551
EUW1_5982849305
EUW1_5982852241
EUW1_5982834160
EUW1_5982385567
EUW1_5982421620
EUW1_5981636284
EUW1_5981413357
EUW1_5981269044
EUW1_5981244521
EUW1_5980174276
EUW1_5980007970
EUW1_5979817728
EUW1_5979851539
EUW1_5979685858
EUW1_5979617134
EUW1_5978185445
EUW1_5978027389
EUW1_5977260536
EUW1_5977193266


In [39]:
# Data on the last 20 games of the account in question
data

{'champion': ['Jhin',
  'Pyke',
  'Ekko',
  'Zoe',
  'Ekko',
  'Veigar',
  'Swain',
  'Yasuo',
  'Swain',
  'Diana',
  'Aatrox',
  'Kassadin',
  'Tristana',
  'Zoe',
  'Velkoz',
  'Malzahar',
  'Malphite',
  'Shen',
  'Elise',
  'Diana'],
 'kills': [11, 20, 5, 12, 12, 6, 12, 6, 3, 10, 5, 6, 8, 9, 7, 7, 10, 4, 7, 16],
 'deaths': [8, 3, 9, 5, 7, 8, 8, 7, 4, 5, 7, 11, 4, 7, 7, 9, 6, 6, 10, 8],
 'assists': [15,
  13,
  9,
  11,
  9,
  3,
  19,
  4,
  6,
  7,
  4,
  6,
  4,
  7,
  8,
  3,
  17,
  16,
  8,
  13],
 'win': [False,
  True,
  False,
  True,
  False,
  False,
  True,
  True,
  False,
  False,
  False,
  True,
  True,
  False,
  True,
  False,
  False,
  True,
  False,
  False]}

The next step requires the "pandas" package \
You may already have this installed, if not: https://pandas.pydata.org/getting_started.html

In [40]:
import pandas as pd # the "as" part just renames it to make it quicker to type

We'll now convert our data dictionary into a dataframe

In [41]:
df = pd.DataFrame(data)
df

champion  kills  deaths  assists    win
0       Jhin     11       8       15  False
1       Pyke     20       3       13   True
2       Ekko      5       9        9  False
3        Zoe     12       5       11   True
4       Ekko     12       7        9  False
5     Veigar      6       8        3  False
6      Swain     12       8       19   True
7      Yasuo      6       7        4   True
8      Swain      3       4        6  False
9      Diana     10       5        7  False
10    Aatrox      5       7        4  False
11  Kassadin      6      11        6   True
12  Tristana      8       4        4   True
13       Zoe      9       7        7  False
14    Velkoz      7       7        8   True
15  Malzahar      7       9        3  False
16  Malphite     10       6       17  False
17      Shen      4       6       16   True
18     Elise      7      10        8  False
19     Diana     16       8       13  False

We'll also convert the previous code into a function for later use

In [42]:
def gather_all_data(puuid, match_ids, mass_region, api_key):
    # We initialise an empty dictionary to store data for each game
    data = {
        'champion': [],
        'kills': [],
        'deaths': [],
        'assists': [],
        'win': []
    }

    for match_id in match_ids:
        print(match_id)

        # run the two functions to get the player data from the match ID
        match_data = get_match_data(match_id, mass_region, api_key)
        player_data = find_player_data(match_data, puuid)

        # assign the variables we're interested in
        champion = player_data['championName']
        k = player_data['kills']
        d = player_data['deaths']
        a = player_data['assists']
        win = player_data['win']

        # add them to our dataset
        data['champion'].append(champion)
        data['kills'].append(k)
        data['deaths'].append(d)
        data['assists'].append(a)
        data['win'].append(win)    
    
    df = pd.DataFrame(data)
    
    return df

In [43]:
df = gather_all_data(puuid, match_ids, mass_region, api_key)

EUW1_5982943551
EUW1_5982849305
EUW1_5982852241
EUW1_5982834160
EUW1_5982385567
EUW1_5982421620
EUW1_5981636284
EUW1_5981413357
EUW1_5981269044
EUW1_5981244521
EUW1_5980174276
EUW1_5980007970
EUW1_5979817728
EUW1_5979851539
EUW1_5979685858
EUW1_5979617134
EUW1_5978185445
EUW1_5978027389
EUW1_5977260536
EUW1_5977193266


In [44]:
df

champion  kills  deaths  assists    win
0       Jhin     11       8       15  False
1       Pyke     20       3       13   True
2       Ekko      5       9        9  False
3        Zoe     12       5       11   True
4       Ekko     12       7        9  False
5     Veigar      6       8        3  False
6      Swain     12       8       19   True
7      Yasuo      6       7        4   True
8      Swain      3       4        6  False
9      Diana     10       5        7  False
10    Aatrox      5       7        4  False
11  Kassadin      6      11        6   True
12  Tristana      8       4        4   True
13       Zoe      9       7        7  False
14    Velkoz      7       7        8   True
15  Malzahar      7       9        3  False
16  Malphite     10       6       17  False
17      Shen      4       6       16   True
18     Elise      7      10        8  False
19     Diana     16       8       13  False

As you can see, the "win" colun is a boolean (True or False) \
To make it easier to use, we'll convert it to an int (1 or 0)

In [45]:
df['win'] = df['win'].astype(int)

In [46]:
df

champion  kills  deaths  assists  win
0       Jhin     11       8       15    0
1       Pyke     20       3       13    1
2       Ekko      5       9        9    0
3        Zoe     12       5       11    1
4       Ekko     12       7        9    0
5     Veigar      6       8        3    0
6      Swain     12       8       19    1
7      Yasuo      6       7        4    1
8      Swain      3       4        6    0
9      Diana     10       5        7    0
10    Aatrox      5       7        4    0
11  Kassadin      6      11        6    1
12  Tristana      8       4        4    1
13       Zoe      9       7        7    0
14    Velkoz      7       7        8    1
15  Malzahar      7       9        3    0
16  Malphite     10       6       17    0
17      Shen      4       6       16    1
18     Elise      7      10        8    0
19     Diana     16       8       13    0

Now we have it as a dataframe, it's much easier to do what we want, such as:

In [47]:
# Find the averages
df.mean(numeric_only=True) # numeric_only stops it trying to average the "champion" column

kills      8.80
deaths     6.95
assists    9.10
win        0.40
dtype: float64

In [48]:
# Get the averages per champion
df.groupby('champion').mean()

kills  deaths  assists  win
champion                             
Aatrox      5.0     7.0      4.0  0.0
Diana      13.0     6.5     10.0  0.0
Ekko        8.5     8.0      9.0  0.0
Elise       7.0    10.0      8.0  0.0
Jhin       11.0     8.0     15.0  0.0
Kassadin    6.0    11.0      6.0  1.0
Malphite   10.0     6.0     17.0  0.0
Malzahar    7.0     9.0      3.0  0.0
Pyke       20.0     3.0     13.0  1.0
Shen        4.0     6.0     16.0  1.0
Swain       7.5     6.0     12.5  0.5
Tristana    8.0     4.0      4.0  1.0
Veigar      6.0     8.0      3.0  0.0
Velkoz      7.0     7.0      8.0  1.0
Yasuo       6.0     7.0      4.0  1.0
Zoe        10.5     6.0      9.0  0.5

In [49]:
# or maybe order your games by amount of kills
df.sort_values('kills')

champion  kills  deaths  assists  win
8      Swain      3       4        6    0
17      Shen      4       6       16    1
2       Ekko      5       9        9    0
10    Aatrox      5       7        4    0
5     Veigar      6       8        3    0
7      Yasuo      6       7        4    1
11  Kassadin      6      11        6    1
15  Malzahar      7       9        3    0
14    Velkoz      7       7        8    1
18     Elise      7      10        8    0
12  Tristana      8       4        4    1
13       Zoe      9       7        7    0
9      Diana     10       5        7    0
16  Malphite     10       6       17    0
0       Jhin     11       8       15    0
4       Ekko     12       7        9    0
3        Zoe     12       5       11    1
6      Swain     12       8       19    1
19     Diana     16       8       13    0
1       Pyke     20       3       13    1

### Adding arguments to the API calls

20 games really isn't enough to evaluate an account, and it also includes ARAMs and Normals which you may not want \
So let's go back to one of our functions and add some arguments

In [50]:
# The original function
def get_match_ids(puuid, mass_region, api_key):
    api_url = (
        "https://" +
        mass_region +
        ".api.riotgames.com/lol/match/v5/matches/by-puuid/" +
        puuid + 
        "/ids?start=0&count=20" + 
        "&api_key=" + 
        api_key
    )
    
    print(api_url)
    
    resp = requests.get(api_url)
    match_ids = resp.json()
    return match_ids      

As you can see from the original function there is already an argument for "count=20" \
So all we need to do is replace that with a variable of our choice

In [51]:
# Updated function where you can set how many games you want
def get_match_ids(puuid, mass_region, no_games, api_key):
    api_url = (
        "https://" +
        mass_region +
        ".api.riotgames.com/lol/match/v5/matches/by-puuid/" +
        puuid + 
        "/ids?start=0" + 
        "&count=" +
        str(no_games) + 
        "&api_key=" + 
        api_key
    )
    
    print(api_url)
    
    resp = requests.get(api_url)
    match_ids = resp.json()
    return match_ids      

In [52]:
no_games = 25 # Leave this as 25 for now, otherwise you may run into issues that we cover shortly!

match_ids = get_match_ids(puuid, mass_region, no_games, api_key)

print(len(match_ids))

https://EUROPE.api.riotgames.com/lol/match/v5/matches/by-puuid/dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA/ids?start=0&count=25&api_key=RGAPI-b77ee1e3-e525-4a6d-8d5c-a9938d45f9d0
25


But let's say we ONLY want data from a certain queue type (I'll use Ranked Solos) \
How do we know what argument to use? 

Go back to the portal, return to MATCH-V5 (on the left) and open "/lol/match/v5/matches/by-puuid/{puuid}/ids" again \

Scroll to the bottom and fill in the form again with the same puuid, region and app \
Except this time, also add "420" to the "queue" input then EXECUTE REQUEST

Copy and paste the REQUEST URL

In [53]:
"https://europe.api.riotgames.com/lol/match/v5/matches/by-puuid/dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA/ids?queue=420&start=0&count=20"

'https://europe.api.riotgames.com/lol/match/v5/matches/by-puuid/dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA/ids?queue=420&start=0&count=20'

As you can see, there's a new argument: "queue=420"

If you wondering where to find all the queue types, you can go here: https://static.developer.riotgames.com/docs/lol/queues.json 

Simply find the queue type you want from the description and replace the "queue" with the queueId 

Let's add this into our function

In [54]:
# Updated function where you can set which queue to take data from
def get_match_ids(puuid, mass_region, no_games, queue_id, api_key):
    api_url = (
        "https://" +
        mass_region +
        ".api.riotgames.com/lol/match/v5/matches/by-puuid/" +
        puuid + 
        "/ids?start=0" + 
        "&count=" +
        str(no_games) + 
        "&queue=" + 
        str(queue_id) + 
        "&api_key=" + 
        api_key
    )
    
    print(api_url)
    
    resp = requests.get(api_url)
    match_ids = resp.json()
    return match_ids      

In [55]:
queue_id = 420

match_ids = get_match_ids(puuid, mass_region, no_games, queue_id, api_key)
match_ids

# Experiment with changing to different queue_ids
# For ones that don't exist, the match_ids should be an empty list

https://EUROPE.api.riotgames.com/lol/match/v5/matches/by-puuid/dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA/ids?start=0&count=25&queue=420&api_key=RGAPI-b77ee1e3-e525-4a6d-8d5c-a9938d45f9d0


['EUW1_5982385567',
 'EUW1_5982421620',
 'EUW1_5981636284',
 'EUW1_5981413357',
 'EUW1_5981269044',
 'EUW1_5981244521',
 'EUW1_5980174276',
 'EUW1_5980007970',
 'EUW1_5979817728',
 'EUW1_5979851539',
 'EUW1_5979685858',
 'EUW1_5979617134',
 'EUW1_5978185445',
 'EUW1_5978027389',
 'EUW1_5973981543',
 'EUW1_5973924798',
 'EUW1_5973820156',
 'EUW1_5973738312',
 'EUW1_5972577481',
 'EUW1_5972563296',
 'EUW1_5972367213',
 'EUW1_5971803328',
 'EUW1_5971718683',
 'EUW1_5971148210',
 'EUW1_5971171984']

### Rate Limits

If you return to the home page of the developer portal \
https://developer.riotgames.com/ \
Under your API key you'll see: RATE LIMITS \
This is currently set to 20 per second & 100 per 2 minutes 

So what happens if we go over it? \
Let's send off the same API request over 100 times to see:

In [56]:
# NOTE, we're sending off the same match ID 200 times, which isn't useful and only for the example
# yours may break before you reach 100, depending how fast you're going through the notebook
for i in range(200):
    print("Attempt:", i)
    match_data = get_match_data(match_id, mass_region, api_key)
    print("Match Duration:", match_data['info']['gameDuration'])

Attempt: 0
Match Duration: 2081
Attempt: 1
Match Duration: 2081
Attempt: 2
Match Duration: 2081
Attempt: 3
Match Duration: 2081
Attempt: 4
Match Duration: 2081
Attempt: 5
Match Duration: 2081
Attempt: 6
Match Duration: 2081
Attempt: 7
Match Duration: 2081
Attempt: 8
Match Duration: 2081
Attempt: 9
Match Duration: 2081
Attempt: 10
Match Duration: 2081
Attempt: 11
Match Duration: 2081
Attempt: 12
Match Duration: 2081
Attempt: 13
Match Duration: 2081
Attempt: 14
Match Duration: 2081
Attempt: 15
Match Duration: 2081
Attempt: 16
Match Duration: 2081
Attempt: 17
Match Duration: 2081
Attempt: 18
Match Duration: 2081
Attempt: 19
Match Duration: 2081
Attempt: 20
Match Duration: 2081
Attempt: 21
Match Duration: 2081
Attempt: 22
Match Duration: 2081
Attempt: 23
Match Duration: 2081
Attempt: 24
Match Duration: 2081
Attempt: 25
Match Duration: 2081
Attempt: 26
Match Duration: 2081
Attempt: 27
Match Duration: 2081
Attempt: 28
Match Duration: 2081
Attempt: 29
Match Duration: 2081
Attempt: 30
Match Du

KeyError: 'info'

The code runs fine and prints the game duration UNTIL the 101st \

So let's inspect what it returned:

In [57]:
match_data

{'status': {'message': 'Rate limit exceeded', 'status_code': 429}}

A status_code of 429, which means we've exceeded our rate limit

The maximum amount of recent games you can request is 100

So, from start to finish, our code currently can send up to 102 requests \
1 to get the summoner names puuid \
1 to get the puuids most 100 recent games \
up to 100 to get data on those games

So, let's update that last function to handle hitting the rate limit

In [58]:
# Original function to get match ID
def get_match_data(match_id, mass_region, api_key):
    api_url = (
        "https://" + 
        mass_region + 
        ".api.riotgames.com/lol/match/v5/matches/" +
        match_id + 
        "?api_key=" + 
        api_key
    )
    
    resp = requests.get(api_url)
    print("Status Code:", resp.status_code) # We can read the status code straight from the "resp" object
    match_data = resp.json()
    return match_data     

First, let's do the loop again except this time printing the status_code from the "resp" object at each stage

NOTE: Depending on how quickly you're going through the notebook, you may get the error before the 101st attempt \
since you are still within the 2 minute rate limit from the previous example

In [59]:
for i in range(120):
    print("Attempt:", i)
    get_match_data(match_id, mass_region, api_key)

Attempt: 0
Status Code: 429
Attempt: 1
Status Code: 429
Attempt: 2
Status Code: 429
Attempt: 3
Status Code: 429
Attempt: 4
Status Code: 429
Attempt: 5
Status Code: 429
Attempt: 6
Status Code: 429
Attempt: 7
Status Code: 429
Attempt: 8
Status Code: 429
Attempt: 9
Status Code: 429
Attempt: 10
Status Code: 429
Attempt: 11
Status Code: 429
Attempt: 12
Status Code: 429
Attempt: 13
Status Code: 429
Attempt: 14
Status Code: 429
Attempt: 15
Status Code: 429
Attempt: 16
Status Code: 429
Attempt: 17
Status Code: 429
Attempt: 18
Status Code: 429
Attempt: 19
Status Code: 429
Attempt: 20
Status Code: 429
Attempt: 21
Status Code: 429
Attempt: 22
Status Code: 429
Attempt: 23
Status Code: 429
Attempt: 24
Status Code: 429
Attempt: 25
Status Code: 429
Attempt: 26
Status Code: 429
Attempt: 27
Status Code: 429
Attempt: 28
Status Code: 429
Attempt: 29
Status Code: 429
Attempt: 30
Status Code: 429
Attempt: 31
Status Code: 429
Attempt: 32
Status Code: 429
Attempt: 33
Status Code: 429
Attempt: 34
Status Code:

We can now catch the error and prevent it from breaking our code 

To do this, we'll use the "time" package, which should be preinstalled

In [60]:
import time

It contains a useful function called "sleep", which simply pauses your function for the given amount of seconds

In [61]:
print("Good Night...")
time.sleep(3)
print("...Good Morning")

# if you decided to change the number to 1000 and now regret it, click Kernel->Interrupt

Good Night...
...Good Morning


In [62]:
# Updated function to sleep until it's succesful
def get_match_data(match_id, mass_region, api_key):
    api_url = (
        "https://" + 
        mass_region + 
        ".api.riotgames.com/lol/match/v5/matches/" +
        match_id + 
        "?api_key=" + 
        api_key
    )
    
    # we need to add this "while" statement so that we continuously loop until it's successful
    while True:
        resp = requests.get(api_url)
        
        # whenever we see a 429, we sleep for 10 seconds and then restart from the top of the "while" loop
        if resp.status_code == 429:
            print("Rate Limit hit, sleeping for 10 seconds")
            time.sleep(10)
            # continue means start the loop again
            continue
            
        # if resp.status_code isn't 429, then we carry on to the end of the function and return the data
        match_data = resp.json()
        return match_data     

In [63]:
# Again, depending on how quick you are running these functions you may immediately hit the sleep
# NOTE: This function takes 2 minutes to run (since it hits the rate limit and then sleeps)
for i in range(101):
    print("Attempt:", i)
    get_match_data(match_id, mass_region, api_key)

Attempt: 0
Rate Limit hit, sleeping for 10 seconds
Attempt: 1
Attempt: 2
Attempt: 3
Attempt: 4
Attempt: 5
Attempt: 6
Attempt: 7
Attempt: 8
Attempt: 9
Attempt: 10
Attempt: 11
Attempt: 12
Attempt: 13
Attempt: 14
Attempt: 15
Attempt: 16
Attempt: 17
Attempt: 18
Attempt: 19
Attempt: 20
Attempt: 21
Attempt: 22
Attempt: 23
Attempt: 24
Attempt: 25
Attempt: 26
Attempt: 27
Attempt: 28
Attempt: 29
Attempt: 30
Attempt: 31
Attempt: 32
Attempt: 33
Attempt: 34
Attempt: 35
Attempt: 36
Attempt: 37
Attempt: 38
Attempt: 39
Attempt: 40
Attempt: 41
Attempt: 42
Attempt: 43
Attempt: 44
Attempt: 45
Attempt: 46
Attempt: 47
Attempt: 48
Attempt: 49
Attempt: 50
Attempt: 51
Attempt: 52
Attempt: 53
Attempt: 54
Attempt: 55
Attempt: 56
Attempt: 57
Attempt: 58
Attempt: 59
Attempt: 60
Attempt: 61
Attempt: 62
Attempt: 63
Attempt: 64
Attempt: 65
Attempt: 66
Attempt: 67
Attempt: 68
Attempt: 69
Attempt: 70
Attempt: 71
Attempt: 72
Attempt: 73
Attempt: 74
Attempt: 75
Attempt: 76
Attempt: 77
Attempt: 78
Attempt: 79
Attempt: 8

### Wrapping it all up

With that complete, we can now wrap all our functions together into one:

In [64]:
def master_function(summoner_name, region, mass_region, no_games, queue_id, api_key):
    puuid = get_puuid(summoner_name, region, api_key)
    match_ids = get_match_ids(puuid, mass_region, no_games, queue_id, api_key)
    df = gather_all_data(puuid, match_ids, mass_region, api_key)
    return df

In [65]:
summoner_name = "Jack J"
region = "euw1"
mass_region = "EUROPE"
no_games = 100
queue_id = 420

In [66]:
# Takes 2 minutes to run
df = master_function(summoner_name, region, mass_region, no_games, queue_id, api_key)

https://euw1.api.riotgames.com/lol/summoner/v4/summoners/by-name/Jack J?api_key=RGAPI-b77ee1e3-e525-4a6d-8d5c-a9938d45f9d0
https://EUROPE.api.riotgames.com/lol/match/v5/matches/by-puuid/dIWQhEOVCY9QT93Jz0EX6YxQIfI3zsyL7tVNmyApKpPmvgNnq5zBu-VKtgGvt2bWzj-fQqd1mYVZMA/ids?start=0&count=100&queue=420&api_key=RGAPI-b77ee1e3-e525-4a6d-8d5c-a9938d45f9d0
EUW1_5982385567
EUW1_5982421620
EUW1_5981636284
EUW1_5981413357
EUW1_5981269044
EUW1_5981244521
EUW1_5980174276
EUW1_5980007970
EUW1_5979817728
EUW1_5979851539
EUW1_5979685858
EUW1_5979617134
EUW1_5978185445
EUW1_5978027389
EUW1_5973981543
EUW1_5973924798
EUW1_5973820156
EUW1_5973738312
EUW1_5972577481
EUW1_5972563296
EUW1_5972367213
EUW1_5971803328
EUW1_5971718683
EUW1_5971148210
EUW1_5971171984
EUW1_5971076871
EUW1_5969828929
EUW1_5969832367
EUW1_5969776040
EUW1_5965112496
EUW1_5964943622
EUW1_5963179116
EUW1_5957333331
EUW1_5941091157
EUW1_5940946566
EUW1_5940994872
EUW1_5940890931
EUW1_5939450518
EUW1_5939362483
EUW1_5937915576
EUW1_5927949

Finally, let's do something useful with the information:

In [67]:
# print some introductory stuff
print("Hello", summoner_name, "of", region.upper()) # upper simply capitalises the region
print("Here are some interesting statistics about your last 100 solo ranked games")

# create a count column
df['count'] = 1 

# the "agg" allows us to get the average of every column but sum the count                                       # see?
champ_df = df.groupby('champion').agg({'kills': 'mean', 'deaths': 'mean', 'assists': 'mean', 'win': 'mean', 'count': 'sum'})

# we reset in the index so we can still use the "champion" column
champ_df.reset_index(inplace=True)

# we limit it to only champions where you've played 2 or more games
champ_df = champ_df[champ_df['count'] >= 2]

# create a kda column
champ_df['kda'] = (champ_df['kills'] + champ_df['assists']) / champ_df['deaths']

# sort the table by KDA, starting from the highest
champ_df = champ_df.sort_values('kda', ascending=False) # ascending determines whether it's highest to lowest or vice-versa

# assign the first row and last row to a variable so we can print information about it
best_row = champ_df.iloc[0] # .iloc[0] simply takes the first row in dataframe
worst_row = champ_df.iloc[-1] # .iloc[-1] takes the last row in a dataframe

print("Your best KDA is on", best_row['champion'], "with a KDA of", best_row['kda'], "over", best_row['count'], "game/s")
print("Your worst KDA is on", worst_row['champion'], "with a KDA of", worst_row['kda'], "over", worst_row['count'], "game/s")

# sort by count instead
champ_df = champ_df.sort_values('count', ascending=False)

# get your most played champ
row = champ_df.iloc[0]

# assign and format the win rate
win_rate = row['win']
win_rate = str(round(win_rate * 100, 1)) + "%"

print("Your highest played Champion is", row['champion'], "with", row['count'], 'game/s', 
     "and an average Win Rate of", win_rate)

# finally, sort by highest kills in a game (note, not using the champ_df groupby anymore but the raw data)
highest_kills = df.sort_values('kills', ascending=False)
row = highest_kills.iloc[0]
print("Your highest kill game was with", row['champion'], "where you had", row['kills'], "kills")


Hello Jack J of EUW1
Here are some interesting statistics about your last 100 solo ranked games
Your best KDA is on Corki with a KDA of 3.6666666666666665 over 2 game/s
Your worst KDA is on Leblanc with a KDA of 1.0 over 2 game/s
Your highest played Champion is Zoe with 9 game/s and an average Win Rate of 22.2%
Your highest kill game was with Veigar where you had 15 kills


### What's else is available?

With that, we've concluded our introduction to the Riot API

To help you think of potential ideas, I'll list a few of the popular APIs that you can use:

- How much Mastery a player has on each Champion
- In-depth game detail for every minute of the game (i.e. how much Gold/XP each player has at 12 minutes)
- In-depth objective and kill data, like who killed who, when and where
- Ranked information, such as their current rank for each queue
- Who is currently in Challenger, Grand Master & Master (& every queue below that too!)
- And much more...!

There's also further advanced topics that I haven't covered in this introduction but you should be aware of:

- Advanced error handling, 429 Rate Limits is just one of many and each requires it's own logic
- Speeding up your code! Eventually, you may decide to build something that requires a lot of data and async/multiprocessing/threading will help
- Building and hosting the frontend of the application
- Getting Riot to approve your application for public use

I've been Jack J from iTero Gaming, I've hoped you've learnt a lot and I'm excited to see what you build!




You can join our Discord to ask questions, show off what you're building or get inspiration:
https://discord.gg/DmSUDGEW6V

You can follow me on Twitter to see what I'm using the Riot API for:
https://twitter.com/JackJGaming

I write a weekly article about statistics/AI/esports/gaming, you can subscribe here: https://medium.com/subscribe/@JackJGaming

Check out the iTero Gaming website to see what we've built:
https://itero.gg/


